huaweicloud/services/acceptance/dns/data_source_huaweicloud_dns_recordsets_test.go

```diff
@@ -0,0 +1,115 @@
+package dns
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceDNSRecordsets_basic(t *testing.T) {
+	rName := "data.huaweicloud_dns_recordsets.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+	name := fmt.Sprintf("acpttest-recordset-%s.com.", acctest.RandString(5))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDatasourceDNSRecordsets_basic(name),
+				Check: resource.ComposeTestCheckFunc(
+					dc.CheckResourceExists(),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.id"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.name"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.zone_id"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.zone_name"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.type"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.ttl"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.records.#"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.status"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.default"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.line_id"),
+					resource.TestCheckResourceAttrSet(rName, "recordsets.0.weight"),
+
+					resource.TestCheckOutput("line_id_filter_is_useful", "true"),
+					resource.TestCheckOutput("status_filter_is_useful", "true"),
+					resource.TestCheckOutput("type_filter_is_useful", "true"),
+					resource.TestCheckOutput("name_filter_is_useful", "true"),
+					resource.TestCheckOutput("recordset_id_filter_is_useful", "true"),
+
+					resource.TestCheckResourceAttr("data.huaweicloud_dns_recordsets.tags_filter", "recordsets.#", "1"),
```

Comment:
add a check: check the key and value of tag

Reply:
The response does not has tag attribute